### DISCUSSION TITLE
Manual/scenes-working-with

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#607] [Notes]
# Manual/scenes-working-with

https://docs.unity3d.com/Manual/scenes-working-with.html

**Comment by MarzLars** [comment]
Just going to copy-paste from the 2019 version of unity docs (I think it is still relevant, if you happen to use DontDestroyOnLoad):
"It is recommended to avoid using DontDestroyOnLoad to persist manager GameObjects that you want to survive across scene loads. Instead, create a manager scene that has all your managers and use SceneManager.LoadScene(<path>, LoadSceneMode.Additive) and SceneManager.UnloadScene to manage your game progress."
Link: https://docs.unity3d.com/2019.3/Documentation/Manual/MultiSceneEditing.html

**Comment by swoobie** [comment]
Loading an existing scene from script isn't described above, but it is supported.

From the manual: "[Create With Unity > Create Gameplay > Scenes > Use scripts to edit multiple scenes](https://docs.unity3d.com/6000.0/Documentation/Manual/scriptmultiplescenes.html)"

> ## Use scripts within the Editor
> 
> When using (or running) scripts within the Editor, use:
> 
> - [Scene struct](https://docs.unity3d.com/ScriptReference/SceneManagement.Scene.html): Available both in the Editor and at runtime, Scene struct contains read-only properties that relate to the scene itself, such as name and asset path.
> - [EditorSceneManager](https://docs.unity3d.com/ScriptReference/SceneManagement.EditorSceneManager.html) API: This class is only available only in the Editor and has several functions to implement all the Multi-Scene editing features described in the pages [Setup multiple scenes](https://docs.unity3d.com/6000.0/Documentation/Manual/setupmultiplescenes.html) and [Bake data in multiple scenes](https://docs.unity3d.com/6000.0/Documentation/Manual/bakemultiplescenes.html).
> - SceneSetup utility class: A utility class that you can use to store information about a scene that is in the Hierarchy window.
> 
> ## Use runtime scripts
> 
> When using scripts at runtime to edit multiple scenes, use the functions in the [SceneManager](https://docs.unity3d.com/ScriptReference/SceneManagement.SceneManager.html) class such as `LoadScene` and `UnloadScene`.
